build.gradle

```diff
@@ -335,7 +335,7 @@ tasks.withType(ScalaCompile) {
         def additionalParameters = ["-Xplugin:" + configurations.scalaCompilerPlugin.asPath,
                                     "-P:scapegoat:dataDir:" + buildDir + "/reports/scapegoat",
                                     "-P:scapegoat:reports:html:xml",
-                                    "-P:scapegoat:disabledInspections:OptionGet:UnsafeTraversableMethods:ComparingUnrelatedTypes:TryGet:EitherGet",
+                                    "-P:scapegoat:disabledInspections:OptionGet:UnsafeTraversableMethods:TryGet:EitherGet:AsInstanceOf:NullAssignment:UnusedMethodParameter:VariableShadowing:NullParameter:MethodReturningAny:IsInstanceOf:EmptyIfBlock:CatchException:EmptyInterpolatedString:UnnecessaryConversion:CatchThrowable:CollectionIndexOnNonIndexedSeq:AvoidSizeNotEqualsZero:ComparingFloatingPointTypes:UnsafeContains:VarCouldBeVal:CatchThrowable:ParameterlessMethodReturnsUnit:SuspiciousMatchOnClassObject:PointlessTypeBounds:BrokenOddness:VarClosure:EmptyTryBlock:RepeatedIfElseBody:BigDecimalDoubleConstructor:EmptyInterpolatedString:SwallowedException:LooksLikeInterpolatedString",
```

Comment:
You added a bunch, but removed `ComparingUnrelatedTypes`. Is there a reason for that?

Reply:
@JustinPihony yes, because ComparingUnrelatedTypes is what this PR is solving. So no need to disable it anymore

Reply:
Ahh, I misread that.